app/app.js

```diff
@@ -0,0 +1,61 @@
+import $ from "jquery";
+import * as tournament from "./tournament/tournament";
+import * as player from "./tournament/player";
+
+var mTournament = new tournament.Tournament();
+
+$(document).ready(function() {
+    $("#buttonAdd").click(addPlayerIfValid);
+    $("#buttonCreate").click(generateTournament);
+});
+
+function addPlayerIfValid() {
+    var info = getPlayerInfo();
+
+    try {
+        var newPlayer = new player.Player(info.name, info.rank);
+        mTournament.addPlayer(newPlayer);
+        updateTable(newPlayer);
+
+    } catch (e) {
```

Comment:
Don´t use exceptions to drive your logic. By reading this, I don´t know who is expected to throw. Use return instead.